nextflow_schema.json

```diff
@@ -289,45 +204,185 @@
                 },
-                "show_hidden_params": {
-                    "type": "boolean",
-                    "fa_icon": "far fa-eye-slash",
-                    "description": "Show all params when using `--help`",
-                    "hidden": true,
-                    "help_text": "By default, parameters set as _hidden_ in the schema are not shown on the command line when a user runs with `--help`. Specifying this option will tell the pipeline to show all parameters."
+                "pipelines_testdata_base_path": {
+                    "type": "string",
+                    "fa_icon": "far fa-check-circle",
+                    "description": "Base URL or local path to location of pipeline test dataset files",
+                    "default": "https://raw.githubusercontent.com/nf-core/test-datasets/",
+                    "hidden": true
                 }
             }
         }
     },
     "allOf": [
         {
-            "$ref": "#/definitions/input_output_options"
+            "$ref": "#/$defs/input_output_options"
         },
         {
-            "$ref": "#/definitions/reference_genome_options"
+            "$ref": "#/$defs/reference_genome_options"
         },
         {
-            "$ref": "#/definitions/institutional_config_options"
+            "$ref": "#/$defs/institutional_config_options"
         },
         {
-            "$ref": "#/definitions/max_job_request_options"
+            "$ref": "#/$defs/generic_options"
+        }
+    ],
+    "properties": {
+        "ont": {
+            "type": "boolean",
+            "description": "ONT reads available?"
         },
-        {
-            "$ref": "#/definitions/generic_options"
+        "hifi": {
+            "type": "boolean",
+            "description": "HiFi reads available?"
+        },
+        "short_reads": {
+            "type": "boolean",
+            "description": "Short reads available?"
+        },
+        "collect": {
+            "type": "boolean",
+            "description": "collect ONT reads into a single file"
+        },
+        "porechop": {
+            "type": "boolean",
+            "description": "run porechop on ONT reads"
+        },
+        "lima": {
+            "type": "boolean",
+            "description": "run lima on HiFi reads?"
+        },
+        "pacbio_primers": {
+            "type": "string",
+            "description": "file containing pacbio primers for trimming with lima"
+        },
+        "trim_short_reads": {
+            "type": "boolean",
+            "description": "trim short reads with trimgalore"
+        },
+        "assembler": {
+            "type": "string",
+            "description": "Assembler to use. Valid choices are: `'hifiasm'`, `'flye'`, or `'flye_on_hifiasm'`. `flye_on_hifiasm` will scaffold flye assembly (ont) on hifiasm (hifi) assembly using ragtag"
+        },
+        "kmer_length": {
+            "type": "integer",
+            "description": "kmer length to be used for jellyfish"
```

Comment:
is there a min and a max value?

Reply:
I assume that there is a range that makes sense, but I think it is nice to give people an option to try weird things?